src/config/configFile/ensureHomeDirFactory.js

```diff
@@ -0,0 +1,30 @@
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+
+const CouldNotCreateHomeDirError = require('../errors/CouldNotCreateHomeDirError')
+
+/**
+ * @param {string} homeDirPath
+ * @return {ensureHomeDir}
+ */
+function ensureHomeDirFactory(homeDirPath = path.resolve(os.homedir(), '.mn')) {
+  /**
+   * @typedef {ensureHomeDir}
+   * @return {string} homeDirPath
+   */
+  function ensureHomeDir() {
+    if (!fs.existsSync(homeDirPath)) {
```

Comment:
If you just check existence in return then it will make this function more readable:

```suggestion
    if (fs.existsSync(homeDirPath)) {
       return homeDirPath;
    }
    
    ...
```

Reply:
Done, is that what you mean?